libs/utils/analysis/frequency_analysis.py

```diff
@@ -296,14 +296,17 @@ def plotCPUFrequencies(self, cpus=None):
             axes.axhline(_avg, color='r', linestyle='--', linewidth=2)
 
             # Set plot limit based on CPU min/max frequencies
-            for cluster,cpus in self._platform['clusters'].iteritems():
-                if cpu_id not in cpus:
-                    continue
-                axes.set_ylim(
-                        (self._platform['freqs'][cluster][0] - 100000)/1e3,
-                        (self._platform['freqs'][cluster][-1] + 100000)/1e3
-                )
-                break
+            if 'clusters' in self._platform:
+                for cluster,cpus in self._platform['clusters'].iteritems():
+                    if cpu_id not in cpus:
+                        continue
+                    freqs = self._platform['freqs'][cluster]
+                    break
+            else:
+                freqs = df['frequency'].unique()
```

Comment:
Why don't we always use this last line without the above checks? Frequencies specified in `platform` should be the the same as the ones you get in the trace or more, in which case it means you do not have events for the other frequencies so you don't plot them.

Reply:
Hmm, I disagree - if you know the platform goes up to 2GHz and the trace was always at 1GHz then you'd like the plot to be around 50% up the axes (`freqs` is used to set the y-axis limits).

Reply:
Mmm, right. Yeah, in that case we can't simplify this code...